torch/lib/TH/generic/THBlas.c

```diff
@@ -18,7 +18,18 @@ TH_EXTERNC void scopy_(int *n, float *x, int *incx, float *y, int *incy);
 TH_EXTERNC void daxpy_(int *n, double *a, double *x, int *incx, double *y, int *incy);
 TH_EXTERNC void saxpy_(int *n, float *a, float *x, int *incx, float *y, int *incy);
 TH_EXTERNC double ddot_(int *n, double *x, int *incx, double *y, int *incy);
+#ifdef __APPLE__
+TH_EXTERNC ffloat cblas_sdot(const int n, const float *x, const int incx, const float *y, const int incy);
+#ifndef THBlas_ACCELERATE_sdot_
```

Comment:
oh i see, so this is an MKL + OSX issue.

Okay in this case, you'd need to create a small test program in-line in FindBLAS.cmake:
https://github.com/pytorch/pytorch/blob/master/torch/lib/TH/cmake/FindBLAS.cmake#L242-L277

and then setan appropriate `-D` in CMakeLists.txt similar to this: https://github.com/pytorch/pytorch/blob/master/torch/lib/TH/CMakeLists.txt#L307-L309